inst/include/dust/interface/dust.hpp

```diff
@@ -193,10 +213,14 @@ SEXP dust_update_state(SEXP ptr, SEXP r_pars, SEXP r_state, SEXP r_step,
     // best thing to take for those? Possibly require an array to
     // disambiguate?
     step = dust::interface::validate_size(r_step, "step");
-    if (!(step.size() == 1 || step.size() == obj->n_particles())) {
+    const size_t len = step.size();
+    if (!(len == 1 || len == obj->n_particles())) {
       cpp11::stop("Expected 'step' to be scalar or length %d",
                   obj->n_particles());
     }
+    if (!std::is_same<T, Dust<typename T::model_type>>::value && len != 1) {
```

Comment:
yikes

Reply:
Why yikes here? this is less hairy than some of the template magic we had before :upside_down_face: